openid-4-verifiable-presentations-1_0.md

```diff
@@ -544,7 +544,7 @@ If the Verifier responds with any HTTP error response, the Wallet MUST terminate
 
 # Response {#response}
 
-A VP Token is only returned if the corresponding Authorization Request contained a `presentation_definition` parameter, a `presentation_definition_uri` parameter, or a `scope` parameter representing a Presentation Definition (#vp_token_request).
+A VP Token MAY be returned if the corresponding Authorization Request contained a `presentation_definition` parameter, a `presentation_definition_uri` parameter, or a `scope` parameter representing a Presentation Definition (#vp_token_request).
```

Comment:
```suggestion
A VP Token MUST be returned when the corresponding Authorization Request contained a `presentation_definition` parameter, a `presentation_definition_uri` parameter, or a `scope` parameter representing a Presentation Definition (#vp_token_request).
```
this is still a MUST. and honestly, re-reading it, the original wording seems right to me - it says return VP Token when the things asking for it are used. @c2bo - what was wrong in your opinion in the original text?

Reply:
> A VP Token is only returned if the corresponding Authorization Request contained a `presentation_definition` parameter, a `presentation_definition_uri` parameter, or a `scope` parameter representing a Presentation Definition (#vp_token_request).

We will return a VP Token whenever a request containing some kind of query was provided. In terms of extensibility of query language, I would understand the old text as we will not return a VP Token unless there was a Presentation Definition - which is not what we want with the new query language.

Reply:
The new query language proposal in https://github.com/openid/OpenID4VP/pull/266/files still returns a VP token, so I think the change actually needed here is the same one as being refined above, i.e. adding something about "or any other parameter defined by this specification that indicates what credential is being requested".

Reply:
yeah, then `is only` is totally fine.